pyscf/tdscf/uhf.py

```diff
@@ -782,6 +1057,181 @@ def vind(xys):
+            e_threshold = numpy.sort(e_ia_a2b)[nstates-1]
+            e_threshold += self.deg_eia_thresh
+
+            idx = numpy.where(e_ia_a2b <= e_threshold)[0]
+            x0 = numpy.zeros((idx.size, nov_a2b))
+            for i, j in enumerate(idx):
+                x0[i, j] = 1  # Koopmans' excitations
+        return x0
+    
+    def init_guess(self, mf, nstates=None, wfnsym=None):
+        if nstates is None: nstates = self.nstates
+        if wfnsym is None: wfnsym = self.wfnsym
+
+        mol = mf.mol
+        mo_energy = mf.mo_energy
+        mo_occ = mf.mo_occ
+        occidxa = numpy.where(mo_occ[0]>0)[0]
+        occidxb = numpy.where(mo_occ[1]>0)[0]
+        viridxa = numpy.where(mo_occ[0]==0)[0]
+        viridxb = numpy.where(mo_occ[1]==0)[0]
+        e_ia_b2a = (mo_energy[0][viridxa,None] - mo_energy[1][occidxb]).T
+        e_ia_a2b = (mo_energy[1][viridxb,None] - mo_energy[0][occidxa]).T
+
+        if wfnsym is not None and mol.symmetry:
+            raise NotImplementedError("UKS Spin Flip TDA/ TDDFT haven't taken symmetry\
+                                      into account.")
+            
+        e_ia_b2a = e_ia_b2a.ravel()
+        e_ia_a2b = e_ia_a2b.ravel()
+        nov_b2a = e_ia_b2a.size
+        nov_a2b = e_ia_a2b.size
+        
+        if self.extype==0:
+            nstates = min(nstates, nov_b2a)
+            e_threshold = numpy.sort(e_ia_b2a)[nstates-1]
+            e_threshold += self.deg_eia_thresh
+
+            idx = numpy.where(e_ia_b2a <= e_threshold)[0]
+            x0 = numpy.zeros((idx.size, nov_b2a))
+            for i, j in enumerate(idx):
+                x0[i, j] = 1  # Koopmans' excitations
+                
+            y0 = numpy.zeros((len(idx),nov_a2b))
+            z0 = numpy.concatenate((x0,y0),axis=1)
+            
+        elif self.extype==1:
+            nstates = min(nstates, nov_a2b)
+            e_threshold = numpy.sort(e_ia_a2b)[nstates-1]
+            e_threshold += self.deg_eia_thresh
+
+            idx = numpy.where(e_ia_a2b <= e_threshold)[0]
+            x0 = numpy.zeros((idx.size, nov_a2b))
+            for i, j in enumerate(idx):
+                x0[i, j] = 1  # Koopmans' excitations
+                
+            y0 = numpy.zeros((len(idx),nov_b2a))
+            z0 = numpy.concatenate((x0,y0),axis=1)
+        return z0
+
+    def kernel(self, x0=None, nstates=None, extype=None):
+        '''SF_TDA diagonalization solver
+        '''
+        cpu0 = (logger.process_clock(), logger.perf_counter())
+        self.check_sanity()
+        self.dump_flags()
+        if extype is None:
+            extype = self.extype
+        else:
+            self.extype = extype
+            
+        if nstates is None:
+            nstates = self.nstates
+        else:
+            self.nstates = nstates
+        log = logger.Logger(self.stdout, self.verbose)
+
+        vind, hdiag = self.gen_vind(self._scf,extype=extype)
+        precond = hdiag
+
+        def pickeig(w, v, nroots, envs):
```

Comment:
Remove this function as it is not used in davidson solver

Reply:
OK. This function will be removed.